src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua

```diff
@@ -278,3 +278,76 @@ function drawImage(image, xPos, yPos)
         end
     end
 end
+
+--- Parses an nft image from a multi-line string
+--
+-- @tparam string image The string containing the raw-nft-image data.
+-- @treturn table The parsed image data, suitable for use with
+-- @{paintutils.drawTextImage}.
+function parseTextImage(image)
+    expect(1, image, "string")
+    local nLine = 1
+    local tImage = {}
+    local sMode = "text"
+    local sBackground = "F"
+    local sForeground = "0"
+    for c in image:gmatch"." do
+        if c == "\n" then
+            nLine = nLine + 1
+        elseif c:byte() == 30 then
+            sMode = "background"
+        elseif c:byte() == 31 then
+            sMode = "foreground"
+        elseif sMode == "background" then
```

Comment:
I'm really not a fan of doing the `sMode` thing. I think a cleaner thing would be something like this:

```lua
local i, len = 1, #image
while i <= len do
  local c = image:sub(i, i)
  if c == "\31" and i < len then
    i = i + 1
    sForeground = image:sub(i, i)
  elseif -- ...
  end

  i = i + 1
end
```